PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/actions/SparkSubmitJobAction.kt

```diff
@@ -90,4 +105,18 @@ class SparkSubmitJobAction : AzureAnAction() {
             model.isClusterSelectionEnabled = true
         }
     }
-}
\ No newline at end of file
+}
+
+open class LivySparkSubmitJobAction(val runConfigurationTypeInstance: ConfigurationType = LivySparkBatchJobRunConfigurationType.getInstance()) : SparkSubmitJobAction() {
+    override fun submitWithPopupMenu(anActionEvent: AnActionEvent): Boolean = false
+    override fun getRunConfigurationFromDataContext(dataContext: DataContext): RunnerAndConfigurationSettings? {
+        val configContext = ConfigurationContext.getFromContext(dataContext)
+        val factory = runConfigurationTypeInstance.configurationFactories[0]
+        val runConfig = factory.createTemplateConfiguration(configContext.project)
```

Comment:
Do we need to customize the config name with a pattern? We can find it next time to avoid creating lots of configs.